src/main/kotlin/platform/sponge/creator/Sponge8ProjectCreator.kt

```diff
@@ -0,0 +1,154 @@
+) : BaseProjectCreator(rootModule, buildSystem) {
+
+    protected fun setupDependencyStep(): SpongeDependenciesSetup {
+        val spongeApiVersion = config.spongeApiVersion
+        return SpongeDependenciesSetup(buildSystem, spongeApiVersion)
+    }
+
+    protected fun setupMainClassStep(): BasicJavaClassStep {
+        return createJavaClassStep(config.mainClass) { packageName, className ->
+            val pluginId = (buildSystem.parent ?: buildSystem).artifactId
+            Sponge8Template.applyMainClass(project, pluginId, packageName, className)
+        }
+    }
+}
+
+class Sponge8MavenCreator(
+    rootDirectory: Path,
+    rootModule: Module,
+    buildSystem: MavenBuildSystem,
+    config: SpongeProjectConfig
+) : Sponge8ProjectCreator<MavenBuildSystem>(rootDirectory, rootModule, buildSystem, config) {
+
+    override fun getSingleModuleSteps(): Iterable<CreatorStep> {
+        val mainClassStep = setupMainClassStep()
+
+        val pomText = SpongeTemplate.applyPom(project)
+
+        return listOf(
+            setupDependencyStep(),
+            BasicMavenStep(project, rootDirectory, buildSystem, config, pomText),
+            mainClassStep,
+            MavenGitignoreStep(project, rootDirectory),
+            BasicMavenFinalizerStep(rootModule, rootDirectory)
+        )
+    }
+
+    override fun getMultiModuleSteps(projectBaseDir: Path): Iterable<CreatorStep> {
+        val depStep = setupDependencyStep()
+        val commonDepStep = CommonModuleDependencyStep(buildSystem)
+        val mainClassStep = setupMainClassStep()
+
+        val pomText = SpongeTemplate.applySubPom(project)
+        val mavenStep = BasicMavenStep(
+            project = project,
+            rootDirectory = rootDirectory,
+            buildSystem = buildSystem,
+            config = config,
+            pomText = pomText,
+            parts = listOf(
+                BasicMavenStep.setupDirs(),
+                BasicMavenStep.setupSubCore(buildSystem.parentOrError.artifactId),
+                BasicMavenStep.setupSubName(config.type),
+                BasicMavenStep.setupInfo(),
+                BasicMavenStep.setupDependencies()
+            )
+        )
+        return listOf(depStep, commonDepStep, mavenStep, mainClassStep)
+    }
+}
+
+class Sponge8GradleCreator(
+    rootDirectory: Path,
+    rootModule: Module,
+    buildSystem: GradleBuildSystem,
+    config: SpongeProjectConfig
+) : Sponge8ProjectCreator<GradleBuildSystem>(rootDirectory, rootModule, buildSystem, config) {
+
+    override fun getSingleModuleSteps(): Iterable<CreatorStep> {
+        val mainClassStep = setupMainClassStep()
+
+        val buildText = Sponge8Template.applyBuildGradle(
+            project,
+            buildSystem,
+            config.spongeApiVersion,
+            config.pluginName,
+            config.mainClass,
+            config.description,
+            config.website,
+            config.authors,
+            config.dependencies
```

Comment:
Might be better to just pass the `config` object in instead of destructuring it here.